tools/rapids-s3-path

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+# A utility script that examines environment variables provided
+# by Jenkins to print out an S3 path where the expected artifact
+# should be.
+# Positional Arguments:
+#   1) a string which determines which artifact path should be returned
+#
+#      if the string is one of "conda_cpp", "conda_python", "wheel_python",
+#      $(arch) will be appended 
+#      
+#      if the string is "conda_python", ${PY_VER/./} will be appended
+#
+#      any other string is unmodified, but has to be non-empty
+#      intended for misc artifacts that are not CPP or Python artifacts
+set -e
+
+repo_name="${GITHUB_REPOSITORY##*/}"
+
+# This function formats the s3 path for an artifact from RAPIDS CI
+#
+# The output format should be one of the following:
+#
+## For PR builds:
+## s3://rapids-downloads/ci/<REPO_NAME>/pull-request/<PR_NUMBER>/<SHORT_HASH>/<REPO_NAME>_<PKG_NAME>
+
+## For branch builds:
+## s3://rapids-downloads/ci/<REPO_NAME>/branch/<BRANCH_NAME>/<SHORT_HASH>/<REPO_NAME>_<PKG_NAME>
+
+## For nightly builds:
+## s3://rapids-downloads/nightly/<REPO_NAME>/<DATE>/<SHORT_HASH>/<REPO_NAME>_<PKG_NAME>
+function format_s3_path {
+  build_type="$1"
+  pkg_name="$2"
+
+  if [ -z "${build_type}" ]; then
+    echo_stderr "Must supply BUILD_TYPE as first argument"
+    return 1
+  fi
+
+  if [ -z "${pkg_name}" ]; then
+    echo_stderr "Must supply PKG_NAME as second argument"
+    return 1
+  fi
+
+  s3_directory_id=""
+  s3_prefix=""
+
+  case "${build_type}" in
+    pull-request)
+      # For PRs, $GITHUB_REF_NAME is like:
+      # 1058/merge
+      s3_directory_id="${GITHUB_REF_NAME%%/*}"
```

Comment:
All pull requests will be copied to a branch in the source repository that has the format `pull-request/<PR_NUMBER>`.

This logic will need to be updated to reflect that. This aligns with our discussions from the CircleCI article below about _Marking code as trusted by pushing to upstream_.

https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/

In essence, technically PR builds will still be branch builds.

Reply:
I changed the logic to the following:
* Add a comment describing the above trusted code situation and how we copy PRs to branches named pull-request in the case of BUILD_TYPE="pull-request"
* Extract the part of the branch name trailing after the `/`, e.g. `BUILD_TYPE="pull-request", GITHUB_REF_NAME="pull-request/1058"` -> `s3_directory_id="1058"`